src/editor/editor.ts

```diff
@@ -0,0 +1,24 @@
+import { Compartment, EditorState, Extension } from "@codemirror/state"
+import { python } from "@codemirror/lang-python"
+import { sql } from '@codemirror/lang-sql'
+
+const languageConf = new Compartment
+
+const autoLanguage = EditorState.transactionExtender.of(tr => {
+    // if (!tr.docChanged) return null
+    let docIsSQL = /^\s*%{1,2}sql/.test(tr.newDoc.sliceString(0, 100))
+    // let stateIsSQL = tr.startState.facet(language) == htmlLanguage
+    // if (docIsSQL == stateIsSQL) return null
+    return {
```

Comment:
do these comments explain logic or is this old code? if old code, I think you can remove

Reply:
I cleaned the editor plugin.